rfc/cdevents-spinnaker.md

```diff
@@ -0,0 +1,149 @@
+EOF 
+```
+A pipeline will be triggered for which Webhook Automated trigger is configured as below:
+![cdevent-consume-webhook.png](cdevent-consume-webhook.png)
+ 
+ Note: An additional validation is required to be added to the above Webhook configuration to include Attribute Constraints for triggering the pipeline only when a specific event type is received.
+ Event type will be included as `ce-type` in the HTTP Header part of the event consumed
+ 
+
+### Produce CDEvents
+
+A new Webhook stage will be configured in a pipeline from which we need to send a CDEvent 
+![cdevent-produce-webhook.png](cdevent-produce-webhook.png)
+
+The CDEvent will be received by a `events-broker` configured as Webhook URL.
+
+With this existing Webhook configuration CDEvents will not be fulfilled as additional CI/CD specific extensions can not be added here. This will be achieved by creating a new Notification type `CDEvents` to produce CDEvents using [Java SDK](https://github.com/cdevents/sdk-java) as part of Goal-2 Implementation.
+
+### Goal-2 : 
+Integrating CDEvents to Spinnaker using Spinnaker Pub/Sub Automated triggers type(google/amazon) and Notifications:
+
+### Consume CDEvents:
+
+CDEvents can be consumed by existing Pub/Sub system types(google/amazon) as CDEvent is build upon CloudEvent and CloudEvent is supported by most of the event providers.
+
+The pipeline can be triggered when a specific event type is received, event type will be included as `ce-type` in the HTTP Header part of the event consumed and it will be configured as one of the Attribute Constraints
+
+![cdevent-consume-pubsub.png](cdevent-consume-pubsub.png)
+
+
+#### Produce CDEvents
+A new Notification type `CDEvents` will be implemented to produce CDEvents using [Java SDK](https://github.com/cdevents/sdk-java).
+
+A Notification will be added at any stage of the pipeline from which we need to send a CDEvent.
+A CDEvent Notification will be sent to the selected `events-broker` and the event type configured when adding new Notification
+
+`CDEvents` can be created using [Java SDK](https://github.com/cdevents/sdk-java) that is available in `CDEvents` [GitHub Repositories](https://github.com/cdevents).
+`Java SDK` used to create various types of events aligning with `CDEvents` vocabulary specification using CloudEventBuilder. 
+
+`CDEvents` can be published to `events-broker` once created from Spinnaker application using CloudEvents HTTP libraries.
+
+Example: A sample service-deployed event published to `events-broker`
+```
+curl -v -d '{"id": "1234", "subject": "event"}' -X POST -H "Ce-Id: HelloSpinnaker" -H "Ce-Specversion: 1.0" -H "Ce-Type: dev.cdevents.service.deployed" -H "Ce-Source: not-sendoff" -H "Content-Type: application/json" "http://<events-broker-url>/default/events-broker"
+```
+
+### Dependencies
+
+The CloudEvent dependencies(io.cloudevents) are required to implement produce/consume events 
+
+## Drawbacks
+
+
```

Comment:
I think we need to highlight that CdEvents is still very early in its life, and things are subject to change. That's a drawback we should highlight. Further, highlighting that if the eventbus does go down, it would bring down the whole system, as before, if a single service of spinnaker failed, like echo, things would still behave somewhat. While we can definitely architect the system in a way that makes the eventbus resilient. It's another drawback.

Reply:
Yes, will add the mentioned drawbacks